crates/rollup/src/driver.rs

```diff
@@ -30,22 +53,61 @@ impl<N: FullNodeComponents> DriverContext for ExExContext<N> {
     }
 }
 
-/// The Rollup Driver entrypoint.
 #[derive(Debug)]
-pub struct Driver<DC: DriverContext> {
-    /// The rollup configuration
-    cfg: Arc<RollupConfig>,
-    /// The context of the node
-    ctx: DC,
+pub struct StandaloneContext;
+
+#[async_trait]
+impl DriverContext for StandaloneContext {
+    async fn recv_notification(&mut self) -> Option<ExExNotification> {
+        // Where should we get notifications from?
+        // We could use the L1 chain provider, but it's already in the Driver struct
+        // and it doesn't work well with the ExExNotification abstraction IMO.
+        todo!()
```

Comment:
My guess would be the standalone context should own another object that listens to new heads in another thread or something like that. This func could then check if there are any heads available

Reply:
Got it, the provider is Clone so this won't be an issue. Also at that point it would be great if it also had "reorg logic" built-in so that we can send a `ExExNotification::ChainReverted` event when a reorg happens, and keep a unified reorg logic inside the driver body, inside the `handle_exex_notification()` method. 

Does that sound too redundant or no?

Reply:
This sounds right to me, but I def think it shouldn't block this pr. I'm curious how the chain reverted events would be detected though

Reply:
@refcell yeah on second thought - it would also be quite redundant to do:

1. use the L1 chain provider to send a new notification to the driver (requires all raw blocks, txs, receipts)
2. inside the driver loop, either fetch from the L1 online provider again or use the in-memory provider

If Kona is already equipped to work well with online providers, this feels a bit clunky imo. Will think a bit more on this